planner/cascades/transformation_rules.go

```diff
@@ -32,12 +39,135 @@ func GetTransformationRules(node plannercore.LogicalPlan) []Transformation {
 }
 
 var transformationMap = map[memo.Operand][]Transformation{
-	/**
-	operandSelect: []Transformation{
-		nil,
+	memo.OperandSelection: {
+		&PushSelDownTableScan{},
+		&PushSelDownTableGather{},
 	},
-	operandProject: []Transformation{
-		nil,
+	memo.OperandDataSource: {
+		&EnumeratePaths{},
 	},
-	*/
+}
+
+// PushSelDownTableScan pushes the selection down to TableScan.
+type PushSelDownTableScan struct {
+}
+
+// GetPattern implements Transformation interface. The pattern of this rule is `Selection -> TableScan`.
+func (r *PushSelDownTableScan) GetPattern() *memo.Pattern {
+	ts := memo.NewPattern(memo.OperandTableScan)
+	return memo.BuildPattern(memo.OperandSelection, ts)
+}
+
+// Match implements Transformation interface.
+func (r *PushSelDownTableScan) Match(expr *memo.ExprIter) bool {
+	return true
+}
+
+// OnTransform implements Transformation interface.
+func (r *PushSelDownTableScan) OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error) {
+	sel := old.GetExpr().ExprNode.(*plannercore.LogicalSelection)
+	ts := old.Children[0].GetExpr().ExprNode.(*plannercore.TableScan)
+	if ts.Handle == nil {
+		return nil, false, false, nil
+	}
+	accesses, remained := ranger.DetachCondsForColumn(ts.SCtx(), sel.Conditions, ts.Handle)
+	if accesses == nil {
+		return nil, false, false, nil
+	}
+	newTblScan := plannercore.TableScan{
+		Source:      ts.Source,
+		Handle:      ts.Handle,
+		AccessConds: ts.AccessConds.Shallow(),
```

Comment:
how about:
```go
AccessConds: append(ts.AccessConds.Shallow(), accesses),
```

Reply:
We are initializing the `AccessConds` field of the new `TableScan` here, so we can just simply use assignment?